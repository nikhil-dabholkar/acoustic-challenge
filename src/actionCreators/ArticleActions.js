import { LOAD_ARTICLE_DETAILS_DATA, LOAD_ALL_ARTICLES_DATA } from '../constants/constants';
const axios = require('axios');

export const fetchSelectedArticleData = (articleId) => {
    return async (dispatch) => {
        const response = await axios.get(`https://content-eu-4.content-cms.com/api/859f2008-a40a-4b92-afd0-24bb44d10124/delivery/v1/content/${articleId}`);
        console.log(response);
        dispatch({
            type: LOAD_ARTICLE_DETAILS_DATA,
            payload: response.data
        })
    }
}

export const fetchAllArticlesData = () => {
    return async (dispatch) => {
        const response = await axios.get(`https://run.mocky.io/v3/8421cb68-a9d9-4575-ad5d-b4396c864231/`);
        console.log(response);
        dispatch({
            type: LOAD_ALL_ARTICLES_DATA,
            payload: response.data
        })
    }
}
