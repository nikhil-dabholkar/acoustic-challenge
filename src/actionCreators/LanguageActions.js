import { SELECT_LANGUAGE_PREFERENCE } from '../constants/constants'

/**
 * This action will be called when user changes the language
 */
export const switchLanguagePreference = (payload) => ({
  type: SELECT_LANGUAGE_PREFERENCE,
  payload: payload
})
