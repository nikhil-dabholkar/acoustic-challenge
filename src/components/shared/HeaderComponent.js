import React from 'react';
import { LanguageSwitcherComponent } from "../index";
import { FormattedMessage } from 'react-intl';
import { Link } from "react-router-dom";
import styles from "./styles/HeaderComponent.module.scss";
import { useLocation } from 'react-router-dom'

function HeaderComponent({ language, switchLanguage }) {
    let location = useLocation();
    
    return (<React.Fragment>
        <nav className="navbar navbar-light bg-white">
            <a className="navbar-brand" href="/">
                <img src={require('../../assets/images/logo.png')} width="100px" className="d-inline-block align-top" alt="" />
            </a>
            <span className={styles.languageSwitcherContainer}>
                <LanguageSwitcherComponent availableLanguages={language.allLanguages} selectedLanguage={language.selectedLanguage} switchLanguage={switchLanguage}></LanguageSwitcherComponent>
            </span>
        </nav>
        <nav className="navbar navbar-expand-lg navbar-light bg-light">
            <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <li className={"nav-item" + ((location.pathname === '/home') ? " active" : "")}>
                        {<Link className="nav-link" to={`/`}><FormattedMessage id="headerComponent.home" defaultMessage="Home"></FormattedMessage></Link>}
                    </li>
                    <li className={"nav-item" + ((location.pathname === '/articles' || location.pathname.startsWith('/articles/')) ? " active" : "")}>
                        {<Link className="nav-link" to={`/articles`}><FormattedMessage id="headerComponent.articles" defaultMessage="Articles"></FormattedMessage></Link>}
                    </li>
                </ul>
            </div>
        </nav>
    </React.Fragment>

    );
}

export default React.memo(HeaderComponent);
