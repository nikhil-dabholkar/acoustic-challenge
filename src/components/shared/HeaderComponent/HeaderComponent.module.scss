.languageSwitcherContainer {
    float: right;
    padding: 0 20px;
}