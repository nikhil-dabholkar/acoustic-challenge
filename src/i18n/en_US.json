{
    "app.header": "English",
    "detailsComponent.dateTooltip": "Date and time is used when ordering a list of design articles. The time will not display in the website.",
    "detailsComponent.bodySectionTooltip": "The design topic, when displayed in the website, will interleave the text chunks and images to create a fluid layout.",
    "detailsComponent.topicTitle": "Topic title",
    "detailsComponent.authorName": "Author name",
    "detailsComponent.date": "Date",
    "detailsComponent.leadImage": "Lead image",
    "detailsComponent.leadImageCaption": "Lead image caption",
    "detailsComponent.LeadImageCredit": "Lead image credit",
    "detailsComponent.profiles": "Profiles",
    "detailsComponent.lead": "Lead",
    "detailsComponent.card": "Card",
    "detailsComponent.default": "Default",
    "detailsComponent.textForBodySection": "Text for body section",
    "listComponent.id": "Id",
    "listComponent.name": "Name",
    "headerComponent.home": "Home",
    "headerComponent.articles": "Articles"
}