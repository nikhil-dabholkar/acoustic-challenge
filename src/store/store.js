import rootReducer from '../reducer/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

export const store = createStore(rootReducer, applyMiddleware(thunk))