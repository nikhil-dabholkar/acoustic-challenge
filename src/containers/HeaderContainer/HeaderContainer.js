import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { switchLanguagePreference } from '../../actionCreators/LanguageActions';
import { HeaderComponent } from '../../components/index';

function HeaderContainer({ language, switchLanguage }) {
    return (<React.Fragment>
        <div data-testid="headerContainer"></div>
        <HeaderComponent language={language} switchLanguage={switchLanguage}></HeaderComponent>
    </React.Fragment>
    );
}

const mapStateToProps = state => {
    return ({
        language: state.languageReducer
    })
};

const mapDispatchToProps = {
    switchLanguage: switchLanguagePreference
}

HeaderContainer.propTypes = {
    language: PropTypes.object.isRequired,
    switchLanguage: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(React.memo(HeaderContainer));
