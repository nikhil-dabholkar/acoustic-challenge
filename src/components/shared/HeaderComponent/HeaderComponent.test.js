import React from 'react';
import HeaderContainer from '../../../containers/HeaderContainer/HeaderContainer';
import HeaderComponent from './HeaderComponent';
import { IntlProvider } from 'react-intl';
import { store } from "../../../store/store";
import { Provider } from 'react-redux';
import { render, cleanup, screen } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import en_US from "../../../i18n/en_US.json";
import { switchLanguagePreference } from "../../../actionCreators/LanguageActions";

afterEach(cleanup);

function renderWithRouter(
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
    return {
        ...render(<Router history={history}>{ui}</Router>),
        history,
    }
}

function rendererWithRouter(
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
    return {
        ...renderer.create(<Router history={history}>{ui}</Router>),
        history,
    }
}

describe('HeaderComponent', () => {
    test('Check if HeaderComponent Renders on /home', async () => {
        const route = '/home';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("headerComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if HeaderComponent Renders on /articles', async () => {
        const route = '/articles';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("headerComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if HeaderComponent Renders on /articles/fa9519d5-0363-4b8d-8e1f-627d802c08a8', async () => {
        const route = '/articles/fa9519d5-0363-4b8d-8e1f-627d802c08a8';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("headerComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if HeaderComponent Renders on /articles/invalid_item', async () => {
        const route = '/articles/invalid_item';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("headerComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if HeaderComponent Renders on /', async () => {
        const route = '/';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("headerComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test("Check if matches HeaderComponent Snapshot", async () => {
        const route = '/';
        const language = {
            selectedLanguage: {"abbr": "EN", "locale": "en", "label": "English", "region": "US"},
            allLanguages: [{"abbr": "EN", "locale": "en", "label": "English", "region": "US"},{"abbr": "PL", "locale": "pl", "label": "Polish", "region": "PL"}]
        };
        const tree = rendererWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderComponent language={language} switchLanguage={switchLanguagePreference}></HeaderComponent></IntlProvider></Provider>, { route }).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
