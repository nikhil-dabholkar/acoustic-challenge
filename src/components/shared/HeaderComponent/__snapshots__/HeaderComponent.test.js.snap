// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Check if matches HeaderComponent Snapshot 1`] = `
<div
  data-testid="headerComponent"
>
  <nav
    className="navbar navbar-light bg-white"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt=""
        className="d-inline-block align-top"
        src="logo.png"
        width="100px"
      />
    </a>
    <span
      className="languageSwitcherContainer"
    >
      <div
        data-testid="languageSwitcherComponent"
      >
        <span>
          English
           | 
        </span>
        <span>
          <a
            className="bx--link"
            href="#"
            onClick={[Function]}
          >
            Polish
          </a>
          
        </span>
      </div>
    </span>
  </nav>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/articles"
            onClick={[Function]}
          >
            Articles
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
